### DISCUSSION TITLE
VersOne.Epub

### DISCUSSION THREAD

**Discussion by Nomeren** [vers-one/EpubReader#107] [Q&A] [answered]
Hi community !
Using VersOne.Epub I have troubles with some ebooks. I can easily open them with Calibre (may be you know this free software) but using VersOne, I have error : System.Xml.XmlException : 'Unexpected end of file has occurred. The following elements are not closed: ncx. Line 17, position 10.'
It seems that the ncx file is not closed or something else. Could you help me to open these files. Tks a lot

**Comment by vers-one** [comment, marked as answer]
NCX is an XML file inside a EPUB 2 book that contains table of contents of the book. The exception message indicates that this file is malformed (most likely, the `<ncx>` tag is not closed at the end of the file). It's possible that Calibre just ignores these errors and doesn't show table of contents for this book.

There are two options here:
1. Fix the NCX file manually (change the extension of the EPUB file to ZIP, edit the NCX file inside, and then change the extension back to EPUB).
2. Edit the OPF file inside the EPUB and remove the references to the NCX file.

**Comment by Nomeren** [comment]
OK
 Tks a lot for your help.

Good job !
